huaweicloud/services/acceptance/cph/resource_huaweicloud_cph_server_test.go

```diff
@@ -0,0 +1,192 @@
+package cph
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+
+	"github.com/chnsz/golangsdk"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/utils"
+)
+
+func getCphServerResourceFunc(cfg *config.Config, state *terraform.ResourceState) (interface{}, error) {
+	region := acceptance.HW_REGION_NAME
+	// getCphServer: Query the CPH instance
+	var (
+		getCphServerHttpUrl = "v1/{project_id}/cloud-phone/servers/{id}"
+		getCphServerProduct = "cph"
+	)
+	getCphServerClient, err := cfg.NewServiceClient(getCphServerProduct, region)
+	if err != nil {
+		return nil, fmt.Errorf("error creating CPH Client: %s", err)
+	}
+
+	getCphServerPath := getCphServerClient.Endpoint + getCphServerHttpUrl
+	getCphServerPath = strings.ReplaceAll(getCphServerPath, "{project_id}", getCphServerClient.ProjectID)
+	getCphServerPath = strings.ReplaceAll(getCphServerPath, "{id}", state.Primary.ID)
+
+	getCphServerOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		OkCodes: []int{
+			200,
+		},
+	}
+	getCphServerResp, err := getCphServerClient.Request("GET", getCphServerPath, &getCphServerOpt)
+	if err != nil {
+		return nil, fmt.Errorf("error retrieving CphServer: %s", err)
+	}
+
+	getCphServerRespBody, err := utils.FlattenResponse(getCphServerResp)
+	if err != nil {
+		return nil, err
+	}
+	statusRaw := utils.PathSearch("status", getCphServerRespBody, nil)
+
+	if fmt.Sprint(statusRaw) == "6" {
+		return nil, golangsdk.ErrDefault404{}
+	}
+
+	return getCphServerRespBody, nil
+}
+
+func TestAccCphServer_basic(t *testing.T) {
+	var obj interface{}
+
+	name := acceptance.RandomAccResourceName()
+	rName := "huaweicloud_cph_server.test"
+
+	rc := acceptance.InitResourceCheck(
+		rName,
+		&obj,
+		getCphServerResourceFunc,
+	)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
```

Comment:
should we add the prePaid check with `acceptance.TestAccPreCheckChargingMode(t)`?

Reply:
no, support prePaid only